finagle-core/src/main/scala/com/twitter/finagle/util/InetSocketAddressUtil.scala

```diff
@@ -1,14 +1,49 @@
 package com.twitter.finagle.util
 
-import java.net.{InetAddress, InetSocketAddress, SocketAddress, UnknownHostException}
+import java.net.{Inet4Address, Inet6Address, InetAddress, InetSocketAddress,
+  NetworkInterface, SocketAddress, SocketException, UnknownHostException}
+import com.twitter.logging.Logger
 
 object InetSocketAddressUtil {
 
   type HostPort = (String, Int)
 
+  private[this] val log = Logger()
+
   private[finagle] val unconnected =
     new SocketAddress { override def toString = "unconnected" }
 
+  private[this] lazy val anyInterfaceSupportsIpV6: Boolean = {
+    try {
+      val interfaces = NetworkInterface.getNetworkInterfaces()
+      while (interfaces.hasMoreElements()) {
+        val iface = interfaces.nextElement()
+        val addresses = iface.getInetAddresses()
+        while (addresses.hasMoreElements()) {
+          val inetAddress = addresses.nextElement()
+          if (inetAddress.isInstanceOf[Inet6Address] && !inetAddress.isAnyLocalAddress() &&
+            !inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress()) {
+            true
```

Comment:
I see this was taken from the patch in the Netty PR. Missing a `return` here for the desired effect.

This is also java-style code. Typically don't use returns in scala code. This could look something like
```
...
import scala.collection.JavaConverters._
...
try {
    val interfaces = NetworkInterface.getNetworkInterfaces().asScala
    interfaces.exists { interface =>
        val addresses = interface.getInetAddresses().asScala
        addresses.exists { inetAddress =>
            inetAddress.isInstanceOf[Inet6Address] && !inetAddress.isAnyLocalAdress() && !inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress()
        }
    }
} catch {
    case NonFatal(_) =>
        // log
        false
}

Reply:
done